servant-server/example/greet.hs

```diff
@@ -25,16 +25,24 @@ instance FromJSON Greet
 instance ToJSON Greet
 
 -- API specification
-type TestApi =
-       -- GET /hello/:name?capital={true, false}  returns a Greet as JSON
-       "hello" :> Capture "name" Text :> QueryParam "capital" Bool :> Get '[JSON] Greet
+
+-- GET /hello/:name?capital={true, false)
+-- returns a Greet as JSON or redirects
+type HelloEndpoint =
+    "hello" :> Capture "name" Text :> QueryParam "capital" Bool :>
+        Verb' 'GET (Redirect :> Result 200 '[JSON] Greet)
```

Comment:
Should we maybe use `:<|>` instead of `:>`, since it's *either* one *or* the other, rather than one *and* the other?

Reply:
It might be just me, but I find both of them equally confusing (not terribly though, I could get used to both being used there). To me, and in very vague terms (i.e don't pay too much attention to this), the `:>` let us chain constraints on the input (request), while `:<|>` lets us "or" constraints, but then neither corresponds to what feels like we're doing under that `Verb`, which would be something like "stating a few properties of the output". `:&` maybe? Not sure.

Reply:
I used `:>` because we already have it.

`:<|>` because of the sum doesn't really make sense, as we can do products too. something like `:><` or `:*` (as something "near enough to" ⊗), but those are also bad as the operator should convey it's fixity (not associative here!), and `:>` does it. `:<` would have "wrong" feeling.

Also having less operators is not a bad thing.

Reply:
> `:<|>` because of the sum doesn't really make sense, as we can do products too. 

What I mean is that *both* `:>` *and* `:<|>` would be allowed inside the `Verb`. The first would correspond to products, the second to sums. That makes a little clearer what the expected handler type would be from the API type, and allows us in the instances to not have to pattern match on individual LHSs.